src/datomock/impl.clj

```diff
@@ -83,4 +86,8 @@
 
 (defn mock-conn*
   [^Database db, ^Log parent-log]
-  (->MockConnection (atom (->MockConnState db [])) (d/next-t db) parent-log (atom nil)))
+  (let [state-agent (-> (agent (->MockConnState db [] nil))
+                        (add-watch ::force-deliver-promise (fn [_ _ old new]
+                                                             (when-not (= old new)
```

Comment:
should use `identical?` here

Reply:
@ChrisBlom yes, it will be equivalent and faster since the `deliver-tx-res` field will not be collection-typed.